frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/VcsCommitMetadataAdapter.java

```diff
@@ -0,0 +1,80 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import java.util.Collections;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.vcs.log.Hash;
+import com.intellij.vcs.log.VcsCommitMetadata;
+import com.intellij.vcs.log.VcsUser;
+import com.intellij.vcs.log.impl.HashImpl;
+import io.vavr.NotImplementedError;
+
+import com.virtuslab.gitmachete.backend.api.ICommitOfManagedBranch;
+
+public class VcsCommitMetadataAdapter implements VcsCommitMetadata {
```

Comment:
Would be nice to include some javadoc for why its needed in the first place... and/or rename the class to include the mention of `Squash` (as IIUC this is/will be solely needed for squash)